lib/rouge/lexers/ttcn3.rb

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class TTCN3 < RegexLexer
+      title "TTCN3"
+      desc "The TTCN3 programming language (ttcn-3.org)"
+
+      tag 'ttcn3'
+      filenames '*.ttcn', '*.ttcn3'
+      mimetypes 'text/x-ttcn3', 'text/x-ttcn'
+
+      def self.keywords
```

Comment:
Variables that are local to the scope of the class will be evaluated and stored every time Rouge is loaded. Since almost all lexers are inapplicable each time a consumer uses Rouge this represents an unnecessary allocation of resources.

Rather than doing this, we recommend using memoisation to put the keywords behind a class function. The downside of this approach is that it affects the way these keywords are used in rules (as you'll see below).